### ISSUE TITLE
Allow Kvrocks to become a replica of Redis

### ISSUE THREAD

**Issue by git-hulk** [apache/kvrocks#1754]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.

### Motivation

We would like to support parsing the RDB file in #1753
and after that, it'd be simple to support Kvrocks as a replica
of the Redis instance.

### Are you willing to submit a PR?

- [x] I'm willing to submit a PR!

**Comment by DMwangnima** [comment]
I am interested in this enhancement. And I'd be appreciated if I could try this assignment.

**Comment by git-hulk** [comment]
Hi @DMwangnima 

Thanks for your interest, assigned to you. But to be noticed, this would depend on issue #1753. But you can propose a solution if you don't mind.

**Comment by DMwangnima** [comment]
Got it. I would do some preliminary preparations first.

**Comment by git-hulk** [comment]
> Got it. I would do some preliminary preparations first.

Thank you! Feel free to raise any discussion or question if you like.

**Comment by git-hulk** [comment]
@DMwangnima You can go ahead since https://github.com/apache/kvrocks/issues/1753 has been merged.

**Comment by DMwangnima** [comment]
@git-hulk I am so sorry that I can not proceed with this issue due to some emergency tasks from my supervisor. Maybe I can find another interesting issue after finishing my tasks.

**Comment by git-hulk** [comment]
@DMwangnima Sure, thanks for your feedback.

**Comment by aleksraiden** [comment]
Now, when we start kvrocks as replice of redis, this error we have:

```
I20240309 22:37:26.429916 72523 replication.cc:423] [replication] Check db name request was sent, waiting for response
E20240309 22:37:26.431612 72523 replication.cc:436] Failed to get the db name, -ERR unknown command '_db_name', with args beginning with:
```

**Comment by git-hulk** [comment]
@aleksraiden Yes, we don't support becoming as a Redis replica for now, so created this issue to track.
